core/src/main/java/org/apache/iceberg/PartitionStatsHandler.java

```diff
@@ -367,7 +440,11 @@ private static PartitionMap<PartitionStats> computeStats(
   }
 
   private static PartitionMap<PartitionStats> collectStatsForManifest(
-      Table table, ManifestFile manifest, StructType partitionType, boolean incremental) {
+      Table table,
+      int version,
```

Comment:
Even though we can deduce the version inside this API, I didn't want to compute the version for each thread. Hence, computed in the caller.